src/models/pipeline-run-task-step/test-data.ts

```diff
@@ -0,0 +1,46 @@
+import { PipelineRunTaskStepModel } from '.';
+
+import { TestDependencyDataType } from '../model-types';
+import { ProgressStatus } from '../progress-status';
+
+/*
+ * TestData array
+ *
+ * Note: First object should always be valid
+ */
+export const testData: TestDependencyDataType<PipelineRunTaskStepModel> = [
+  {
+    __testDescription: 'Valid full object',
+    name: 'some-task-step',
+    started: new Date('2022-05-10T14:31:23Z'),
+    ended: new Date(),
+    status: ProgressStatus.Succeeded,
+  },
+  {
+    __testDescription: 'Valid partial object',
+    name: 'some-task-step',
+    started: new Date('2022-05-10T14:31:23Z'),
+    status: ProgressStatus.Queued,
+  },
+  {
+    __testDescription: 'Invalid full object',
+    __testIsInvalidSample: true,
+    name: 'some-task-step',
+    started: new Date('2022-05-10T14:31:23Z'),
+    ended: new Date(),
+    status: ProgressStatus.Succeeded,
+  },
+  {
+    __testDescription: 'Invalid partial object',
+    __testIsInvalidSample: true,
+    name: '',
+    started: new Date('2022-05-10T14:31:23Z'),
```

Comment:
remove optional object in 'Invalid partial object' test

Note that this may cause this test to fail. The solution to this is to make one of the other objects invalid
for example `name: { object: 'name' } as unknown as string,`

```suggestion
```